src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -249,17 +249,14 @@ public Map<UUID, Set<IamRole>> listAuthorizedSnapshots(AuthenticatedUserRequest
    * @param userReq authenticated user
    * @return accessible snapshot UUIDs and the IamRoles dictating their accessibility, established
    *     indirectly by any linked RAS passport
+   * @throws ParseException
    */
-  public Map<UUID, Set<IamRole>> listRasAuthorizedSnapshots(AuthenticatedUserRequest userReq) {
+  public Map<UUID, Set<IamRole>> listRasAuthorizedSnapshots(AuthenticatedUserRequest userReq)
+      throws ParseException {
     Map<UUID, Set<IamRole>> idsAndRoles = new HashMap<>();
-    try {
-      List<RasDbgapPermissions> permissions = ecmService.getRasDbgapPermissions(userReq);
-      List<UUID> uuids = snapshotDao.getAccessibleSnapshots(permissions);
-      Set<IamRole> roles = Set.of(IamRole.READER);
-      uuids.forEach(uuid -> idsAndRoles.put(uuid, roles));
-    } catch (ParseException ex) {
-      logger.error("Error parsing RAS passport", ex);
-    }
+    List<RasDbgapPermissions> permissions = ecmService.getRasDbgapPermissions(userReq);
+    List<UUID> snapshotIds = snapshotDao.getAccessibleSnapshots(permissions);
+    snapshotIds.forEach(snapshotId -> idsAndRoles.put(snapshotId, Set.of(IamRole.READER)));
     return idsAndRoles;
```

Comment:
Streaming is possible here
```suggestion
    return snapshotDao.getAccessibleSnapshots(permissions)
        .stream()
        .collect(Collectors.toMap(Function.identity(), id -> Set.of(IamRole.READER)));
```

Reply:
I had considered this initially, but felt like streaming made it less readable.  However, hearing from both you and Muscles convinces me to put it back in.